### PR TITLE
Add libcurl external repository

### PR DESCRIPTION
**Pull request by schroederc**
third_party/curl.BUILD was borrowed from https://github.com/tensorflow/tensorflow/blob/bb890ea5fa4c8a2100a295e93c5bf37b3c88b55a/third_party/curl.BUILD

### CODE REVIEW

File: WORKSPACE [hunk 1]
```diff
@@ -38,6 +38,17 @@ bind(
     actual = "@com_google_code_gson_gson//jar",
 )
 
+new_http_archive(
+    name = "curl",
```
Comment (shahms) [inline WORKSPACE]: `se_haxx_curl` ?

File: third_party/curl.BUILD [hunk 2]
```diff
@@ -0,0 +1,679 @@
...
+        "#endif",
+        "",
+        "#endif  // EXTERNAL_CURL_INCLUDE_CURL_CONFIG_H_",
+        "EOF",
+    ]),
+)
+
+config_setting(
+    name = "ios",
+    values = {"crosstool_top": "//tools/osx/crosstool:crosstool"},
```
Comment (shahms) [inline third_party/curl.BUILD]: This isn't a valid target

File: third_party/curl.BUILD [hunk 3]
```diff
@@ -0,0 +1,679 @@
...
+)
+
+config_setting(
+    name = "windows",
+    values = {"cpu": "x64_windows_msvc"},
+)
+
+config_setting(
+    name = "android",
+    values = {"crosstool_top": "//external:android/crosstool"},
```
Comment (shahms) [inline third_party/curl.BUILD]: I don't think this is either

File: WORKSPACE (hunk 1 above)
Comment (schroederc) [reply WORKSPACE]: ✓

File: third_party/curl.BUILD (hunk 2 above)
Comment (schroederc) [reply third_party/curl.BUILD]: ✓

File: third_party/curl.BUILD (hunk 3 above)
Comment (schroederc) [reply third_party/curl.BUILD]: ✓

**Pull request merged**